lib/logstash/inputs/kafka.rb

```diff
@@ -98,7 +104,7 @@ def register
       options[:reset_beginning] = 'from-beginning'
     end # if :reset_beginning
     @kafka_client_queue = SizedQueue.new(@queue_size)
-    @consumer_group = Kafka::Group.new(options)
+    @consumer_group = @kafka_group_class.new(options)
```

Comment:
another way instead of subclassing `LogStash::Inputs::Kafka` would have been to trap the create_consumer_group call with something like:

```ruby
it 'should retrieve event from kafka' do
  kafka = LogStash::Inputs::Kafka.new(kafka_config)
  expect(kafka).to receive(:create_consumer_group) do |options|
    TestKafkaGroup.new(options)
  end
  kafka.register

  logstash_queue = Queue.new
```